drivers/spi/spi_mspm0.c

```diff
@@ -0,0 +1,296 @@
+	} else {
+		DL_SPI_disableLoopbackMode(cfg->base);
+	}
+
+	DL_SPI_enable(cfg->base);
+
+	/* save config so it can be reused
+	 * also it's required for the lock owner to work */
+	data->ctx.config = spi_cfg;
+
+	return 0;
+}
+
+static bool spi_mspm0_transfer_ongoing(struct spi_mspm0_data *data)
+{
+	return spi_context_tx_on(&data->ctx) || spi_context_rx_on(&data->ctx);
+}
+
+static void spi_mspm0_frame_tx(const struct device *dev)
+{
+	struct spi_mspm0_data *data = dev->data;
+	const struct spi_mspm0_config *cfg = dev->config;
+	struct spi_context *ctx = &data->ctx;
+
+	/* nop tx if no data is expected */
+	uint32_t tx_frame = 0;
+	if (spi_context_tx_buf_on(&data->ctx)) {
+		if (data->dfs == 1) {
+			tx_frame = UNALIGNED_GET((uint8_t *)(ctx->tx_buf));
+		} else if (data->dfs == 2) {
+			tx_frame = UNALIGNED_GET((uint16_t *)(ctx->tx_buf));
+		} else {
+			tx_frame = UNALIGNED_GET((uint32_t *)(ctx->tx_buf));
+		}
+	}
+
+	DL_SPI_transmitDataBlocking32(cfg->base, tx_frame);
+	while (DL_SPI_isBusy(cfg->base)) {
+		/* wait for tx fifo to be sent */
+	}
+
+	spi_context_update_tx(ctx, data->dfs, 1);
+}
+
+static void spi_mspm0_frame_rx(const struct device *dev)
+{
+	struct spi_mspm0_data *data = dev->data;
+	const struct spi_mspm0_config *cfg = dev->config;
+	struct spi_context *ctx = &data->ctx;
+
+	/* rx fifo will contain 0s if there is no data available, so make sure to clean out those */
+	const uint32_t rx_frame = DL_SPI_receiveDataBlocking32(cfg->base);
+	if (spi_context_rx_buf_on(&data->ctx)) {
+		if (data->dfs == 1) {
+			UNALIGNED_PUT(rx_frame, (uint8_t *)ctx->rx_buf);
+		} else if (data->dfs == 2) {
+			UNALIGNED_PUT(rx_frame, (uint16_t *)ctx->rx_buf);
+		} else {
+			UNALIGNED_PUT(rx_frame, (uint32_t *)ctx->rx_buf);
+		}
+
+		spi_context_update_rx(ctx, data->dfs, 1);
+	}
+}
+
+static void spi_mspm0_start_transfer(const struct device *dev)
+{
+	struct spi_mspm0_data *data = dev->data;
+
+	spi_context_cs_control(&data->ctx, true);
+
+	while (spi_mspm0_transfer_ongoing(data)) {
+		spi_mspm0_frame_tx(dev);
+		spi_mspm0_frame_rx(dev);
+	}
+
+	spi_context_cs_control(&data->ctx, false);
+	spi_context_complete(&data->ctx, dev, 0);
+}
+
+static int spi_mspm0_transceive(const struct device *dev, const struct spi_config *spi_cfg,
+				const struct spi_buf_set *tx_bufs,
+				const struct spi_buf_set *rx_bufs)
+{
+	struct spi_mspm0_data *data = dev->data;
+	spi_context_lock(&data->ctx, false, NULL, NULL, spi_cfg);
+
+	int ret = spi_mspm0_configure(dev, spi_cfg);
+	if (ret != 0) {
+		spi_context_release(&data->ctx, ret);
+		return ret;
+	}
+
+	spi_context_buffers_setup(&data->ctx, tx_bufs, rx_bufs, data->dfs);
+	spi_mspm0_start_transfer(dev);
+
+	ret = spi_context_wait_for_completion(&data->ctx);
+	spi_context_release(&data->ctx, ret);
+
+	return ret;
+}
+
+static int spi_mspm0_release(const struct device *dev, const struct spi_config *config)
+{
+
+	struct spi_mspm0_data *data = dev->data;
+	const struct spi_mspm0_config *cfg = dev->config;
```

Comment:
Given the function only uses this once and it almost conflicts with an argument, I'd just type it all out when used.